x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/approximate/Reliable.java

```diff
@@ -0,0 +1,249 @@
+        BooleanBlock.Builder builder,
+        @Position int position,
+        DoubleBlock estimatesBlock,
+        IntBlock trialCountBlock,
+        IntBlock bucketCountBlock
+    ) {
+        if (trialCountBlock.getValueCount(position) != 1 || bucketCountBlock.getValueCount(position) != 1) {
+            builder.appendNull();
+            return;
+        }
+        double[] estimates = new double[estimatesBlock.getValueCount(position)];
+        for (int i = 0; i < estimatesBlock.getValueCount(position); i++) {
+            estimates[i] = estimatesBlock.getDouble(estimatesBlock.getFirstValueIndex(position) + i);
+        }
+        int trialCount = trialCountBlock.getInt(trialCountBlock.getFirstValueIndex(position));
+        int bucketCount = bucketCountBlock.getInt(bucketCountBlock.getFirstValueIndex(position));
+        builder.appendBoolean(computeReliable(estimates, trialCount, bucketCount));
+    }
+
+    @Evaluator(extraName = "Int")
+    static void process(
+        BooleanBlock.Builder builder,
+        @Position int position,
+        IntBlock estimatesBlock,
+        IntBlock trialCountBlock,
+        IntBlock bucketCountBlock
+    ) {
+        if (trialCountBlock.getValueCount(position) != 1 || bucketCountBlock.getValueCount(position) != 1) {
+            builder.appendNull();
+            return;
+        }
+        double[] estimates = new double[estimatesBlock.getValueCount(position)];
+        for (int i = 0; i < estimatesBlock.getValueCount(position); i++) {
+            estimates[i] = estimatesBlock.getInt(estimatesBlock.getFirstValueIndex(position) + i);
+        }
+        int trialCount = trialCountBlock.getInt(trialCountBlock.getFirstValueIndex(position));
+        int bucketCount = bucketCountBlock.getInt(bucketCountBlock.getFirstValueIndex(position));
+        builder.appendBoolean(computeReliable(estimates, trialCount, bucketCount));
+    }
+
+    @Evaluator(extraName = "Long")
+    static void process(
+        BooleanBlock.Builder builder,
+        @Position int position,
+        LongBlock estimatesBlock,
+        IntBlock trialCountBlock,
+        IntBlock bucketCountBlock
+    ) {
+        if (trialCountBlock.getValueCount(position) != 1 || bucketCountBlock.getValueCount(position) != 1) {
+            builder.appendNull();
+            return;
+        }
+        double[] estimates = new double[estimatesBlock.getValueCount(position)];
+        for (int i = 0; i < estimatesBlock.getValueCount(position); i++) {
+            estimates[i] = estimatesBlock.getLong(estimatesBlock.getFirstValueIndex(position) + i);
+        }
+        int trialCount = trialCountBlock.getInt(trialCountBlock.getFirstValueIndex(position));
+        int bucketCount = bucketCountBlock.getInt(bucketCountBlock.getFirstValueIndex(position));
+        builder.appendBoolean(computeReliable(estimates, trialCount, bucketCount));
+    }
+
+    public static boolean computeReliable(double[] estimates, int trialCount, int B) {
```

Comment:
I'd leave a comment here:

This is a two-tailed p-value test vs the normal distribution using 0.05 significance. We test the skew (which should be zero) and excess kurtosis (which should also be zero) under the null hypothesis. We use a majority vote for the trial count. This is applied to the same samples we use for estimating the confidence interval.